content/blog/the-importance-of-ubiquitous-language/index.md

```diff
@@ -0,0 +1,188 @@
+
+## The importance
+So why is it important that you all speak the same language?  For decades, huge organizations have adopted the "waterfall" mentality of distilling a business domain into a set of requirements by requirements writers, which were then translated by architects into high level architectures, which were then translated by senior software engineers into lower level software designs, which were then translated and implemented by software engineers on a number of teams throughout the organization.  Ultimately turning the software development process into a huge game of telephone.
+
+!["Hi there you" -> "High fair food" -> "I know you" -> "I know kung fu"](./images/telephone.png)
+
+Some domain terms may have made it from the requirements into the software but a large majority didn't.  This model primarily breaks down when there is an expectation of bidirectional, or skip-level, communication.  The reality is that software development isn't as simple as requirements to architecture to design to code. In order to build software, especially in an "agile" or "lean" manner, especially on a balanced team, you empower all roles to engage with the business. It isn't just the PMs responsibility to learn and understand the business context, but everyone's. In order for this to happen effectively, individuals must communicate.  That communication goes a lot more smoothly if the words that the engineers are using have the same meaning as those translating the business domain to write stories.
+
+Let's use an example to illustrate what you are talking about.  Say that your team is writing a shopping application and your product manager has a user interview with someone from the business and write a single story that says something along the lines of:
+
+> As a customer, I want to add products to my shopping cart, so that I can continue shopping until I'm ready to check out.
+
+The architect may then make some high level architecture decisions:
+* A shopping cart should be implemented as a list of items that are comprised of a product and quantity
+* A customer is an authenticated user
+* Once the customer checks out, the items in the shopping cart should be removed
+
+Now, being in the hands of the developer, they write something like this:
+
+```java
+class Item {
+    Product product;
+    int quantity;
+}
+
+class ItemList {
+    List<Item> itemList;
+    
+    void add(Item item) {
+        //add to list
+    }
+    
+    void remove(Item item) {
+        //remove from list
+    }
+    
+    void removeAll() {
+        //removes all items from list
+    }
+}
+```
+
+Although the implementation is technically correct, the terminology they used when writing the code has now given the developers a different vocabulary than other members of the team.  Now you have introduced some translations that need to happen:
+
+* Product <--> Item
+* Cart <--> ItemList
+* ??? <--> Product
+* "Checkout" <--> "Remove all items from cart"
+
```

Comment:
Nice! I like seeing this specific example. It makes it much more concrete what point you're trying to get across.